src/main/scala/org/apache/daffodil/DaffodilPlugin.scala

```diff
@@ -57,11 +60,55 @@ object DaffodilPlugin extends AutoPlugin {
   }
 
   override lazy val projectSettings: Seq[Setting[_]] = Seq(
+    /**
+     * Default Daffodil version
+     */
+    daffodilVersion := "3.6.0",
+
+    /**
+     * Add Daffodil and version specific test dependencies
+     */
+    libraryDependencies ++= {
+      // Seq of 2-tuples, where each tuple is a Seq of version specifiers and a list of
+      // dependencies to add if the daffodilVersion matches all of those specifiers. If the
+      // version specifier Seq is empty, the associated dependencies are added regardless of
+      // Daffodil version
+      val versionedDeps = Seq(
+        // always add Daffodil and junit test dependencies
+        Nil -> Seq(
+          "org.apache.daffodil" %% "daffodil-tdml-processor" % daffodilVersion.value % "test",
+          "junit" % "junit" % "4.13.2" % "test",
+          "com.github.sbt" % "junit-interface" % "0.13.2" % "test",
+        ),
+        // Add log4j with older versions of Daffodil to silence warnings about missing loggers
+        Seq(">=3.2.0", "<=3.4.0") -> Seq(
+          "org.apache.logging.log4j" % "log4j-core" % "2.20.0" % "test",
+        ),
+      )
```

Comment:
Needs a scripted test to make sure these settings are actually set correctly. the saved-parsers-01 test doesn't actually used any of these settings to save a parser.

Reply:
Agreed, should we add the scripted test in this PR or?